### ISSUE TITLE
OMV setup using Deluge in a Docker and everything else locally in OMV

### ISSUE THREAD

**Issue by ParadingLunatic** [clinton-hall/nzbToMedia#1061]
I'm trying to figure out how I'm going to set this up.  Mainly it's folder locations that I'm really confused about.  I'm kind of a Linux newbie.  I know enough to get around and get things done but there's still quite a bit that go way over my head.

The way my configuration is set up, I have sabnzb, couchpotato, and sickrage all running directly on my OpenMediaVault NAS using their plugins.  The Deluge plugin is an older version (1.3.5) so I created a Docker image/container running Deluge 1.3.12 instead of running the older Deluge directly on the system.  The way I have the docker image setup is to attach (map) my NAS Download folder, script folder, config (specific for that Deluge container) to specific locations within the Docker directly off of root.  So for example:

NAS Location ---->  Docker Location

/media/abcdefgxxxxxx/Downloads    ---->   /Downloads
/media/abcdefgxxxxxx/Home/username/scripts ----->  /Scripts
/media/abcdefgxxxxxx/Docker/delugeconfig  -----> /config

I have my downloaders (sabnzb and deluge) to place incompletes/inprocess downloads to /media/abcd.../Downloads/incomplete

Complete TV shows to go .../Downloads/completetv
Complete movies go to ..../Downloads/completemovie
Uncategorized go to .../Downloads/complete

I have all of the nzbtomedia scripts in the scripts folder which is the same obviously between the NAS and the Docker, but the absolute folder is different between the two of course

As you can tell, the absolute folders are different.  Because of this, I'm guessing I'll need to place another copy of the nzbtomedia scripts on each system instead of sharing it.  The whole mount points thing confuses me a little.  So in my case I'm guessing it would look something like this for mount points:

On NAS
mount_points=/media/abcdefgxxxxxxx/Downloads,/Downloads

On Docker
mount_points=/Downloads,/media/abcdefgxxxxxxx/Downloads

What would I put for the watch_dir for Sickbeard/CouchPotato on either server or do I just leave it blank on both in both locations?  Or would I set the watch path for them on the system where SB/CP is on and on the remote server set the remote_path = 1?

**Comment by clinton-hall** [comment]
ok, basically, for SB, CP, SABnzbd the directories are all the same, so no need for mount points there.

In general, For watch directories, these are only needed for manual runs... so again this is outside of docker
```
/media/abcdefgxxxxxxx/Downloads/completetv
/media/abcdefgxxxxxxx/Downloads/completemovie
```

now the only place you need to worry about mount points is with deluge.
When deluge launches the script it will try and tell SickBeard/CouchPotato to scan /Downloads. The problem being the files are actually in /media/abcdefgxxxxxxx/Downloads/

So the correct mount_points for the script triggered by deluge (docker) would be
`mount_points=/Downloads,/media/abcdefgxxxxxxx/Downloads`

The problem here is replacing `/Downloads` with `/media/abcdefgxxxxxxx/Downloads` could cause issues if `/media/abcdefgxxxxxxx/Downloads` becomes `/media/abcdefgxxxxxxx/media/abcdefgxxxxxxx/Downloads`

So you certainly don't want SABnzbd and Deluge to launch the same script.... you wand mount points defined ONLY in the script run by Deluge.

OR... the safest way would be to sym-link /media/abcdefgxxxxxxx/Downloads to /Downloads in your NAS root. `ln -sf /media/abcdefgxxxxxxx/Downloads /Downloads` and not use any mount_points.

This was, even when Deluge tells SB/CP to look in /Downloads this will exists and link to /media/abcdefgxxxxxxx/Downloads

**Comment by ParadingLunatic** [comment]
Ahhh I didn't even think about sym-linking to /Downloads.  I fooled around with it a bit yesterday before reading this.  Basically by making a second copy of the script on the Deluge docker and setting the mount points as you mentioned.  I tried a torrent download via CP but it failed to process.  I also got a little confused with some of the other configuration but I might be able to figure it out.  It's mainly with configuring the Torrent download. I'll give it another shot later tonight with the sym-links as well as changing up my Download folder structure a little to follow more along the lines of the instructions.

**Comment by ParadingLunatic** [comment]
Ok so I have another question.  The final location where CP and and SR place the videos after post processing is completed is located on the NAS at /media/abcdefgxxxxxxx/Videos/TV and .../Movies.  When switching to nzbtomedia to do the post processing on the Deluge Docker, would the Deluge container need to have access to that location, because currently it does not since CP and SR are doing the post processing?  I have a sneaking suspicion it does which means I'll need to either rebuild the container or copy it to a new configuration and add that volume and possibly create another sym-link.

**Comment by clinton-hall** [comment]
No. The docker does not require access to the final library...

Once the script has finished doing some linking and checks etc, it actually sends the current location to SR/CP to do the rename... so deluge never needs to know where these end up.

The only case where there may be issues is with the option to have Torrents link=move-sym. Once SR/CP have moved and linked the final file, the script will follow the symlinks and remove all intermediate steps so that the seeding files (links) point directly to the library media files. In this case, deluge won't be able to seed since these links point outisde of docker...

So, the only limitation I can think of here is that if you want deluge to continue seeding, you need use  [Torrent] link = hard, or link = copy. In SR/CP you need to have these set to move.
Or you might be able to use [Torrent] link = sym, and SR/CP set to sym-link files.

I would try link = hard first if you need seeding...

**Comment by ParadingLunatic** [comment]
Well, I got it mostly working.  Go figure after all of this, Deluge works perfectly but I'm having issues so far with Sabnzbd.  (I attribute understanding the Deluge stuff to this post https://github.com/clinton-hall/nzbToMedia/issues/96) I tried grabbing something with SR, it passed it to sabnzbd just fine.  Once it came time to run the script, the script errored.  I checked the sabnzbd.log file, but there's nothing in there regarding the error and the nzbtomedia.log didn't update.  The only place I see it is on the webui.  This is the most I can get from the WebUI:

Exception TypeError: "'NoneType' object is not callable" in <bound method PosixProc

I'm pretty lost as to what the issue is.  I'm also pretty tired as it's almost 2am.  If needed I can copy/paste the contents of my config's etc.
